.circleci/config.yml

```diff
@@ -0,0 +1,48 @@
+# 1. Set your GitHub default branch to dev/develop
+# 2. In circleci project settings->advanced set:
+#    - GitHub Status Updates (this should be on already)
+#    - Only build pull requests
+#    - Auto-cancel redundant builds
+#    - Free and Open Source (this should be on already)
+# 3. If existing project with quay.io, remove the quay build triggers.
+#    - as circleci is pusing the image
+#    - reevaluate when/if quay.io adds ability to authenticate pulls
+
+version: 2.1
+jobs:
+  build:
+    environment:
+      IMAGE_NAME: quay.io/wtsicgp/gripss
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    steps: 
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: |
+            set -e
+            BRANCH_OR_TAG="${CIRCLE_TAG:-$CIRCLE_BRANCH}"
+            CLEAN_BRANCH=$(echo $BRANCH_OR_TAG | tr / -)
+            echo "$DOCKERHUB_PASSWORD" | docker login -u "$DOCKERHUB_USER" --password-stdin
+            echo "$QUAYIO_PASSWORD" | docker login -u "$QUAYIO_USER" --password-stdin quay.io
+            docker build -t $IMAGE_NAME:$CLEAN_BRANCH .
+            echo 'Verify program(s) from this repo'
+            docker run -t --rm $IMAGE_NAME:$CLEAN_BRANCH bash -c "java com.hartwig.hmftools.gripss.GripssApplicationKt  | grep 'Missing required options' && echo 'gripss OK'"
```

Comment:
Does `GripssApplicationKt` not have a version number we can test for that doesn't give a non-zero exit code?

Reply:
No there is no version nor help option. Plus it throws a 1 exit status with no arguments or wrong arguments. However using this grep trick seems to produce the desired behaviour.